java-agent/src/test/groovy/io/opentelemetry/auto/AgentLoadedIntoBootstrapTest.groovy

```diff
@@ -31,4 +33,23 @@ class AgentLoadedIntoBootstrapTest extends Specification {
       , [:]
       , true) == 0
   }
+
+  def "AgentBootstrap is loaded not from dd-java-agent.jar"() {
```

Comment:
I don't understand what this test does and how it is related to changes in AgentBootstrap. Extra documentation maybe?

Reply:
PR that added this change: https://github.com/DataDog/dd-trace-java/pull/1561

Reply:
I still don't understand. `AgentBootstrap` was changed to throw a RuntimException if it detects that it is being loaded from the unknown jar. Test seems to create some new jar, runs it and verifies that exit code is 0. Does not make sense to me. It should throw an exception, right? If not, then at least one developer (me) got confused by this test :) Thus, it needs explanation.

Reply:
You are right: it might be confusing. 

The actual condition check of this test here:
`https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/553/files#diff-c9d742d3b84ae27c73d8b33c042b0c0cR20`
Test is checking that application level class was not loaded by bootstrap classloader. Check for exit code 0, means app exits w/o throwing that exception in user code. 

No exception and exit code 0 in test, because we don't crash user app, if user wrongly packed (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/553/files#diff-81978fbee285440c149c08f06b126fe8R40) our AgentBootstrap (and maybe some other dd agent classes) into their own "uber.jar" (fat jar). We just disable dd agent with ERROR log in this case, and keep user app running.




The code change checks that we loading dd-java-agent classes only from dd-java-agent.jar . If we detect that our dd-java-agent classes were wrongly packed into "uber.jar" (fat jar) by build system (Spring or some other framework) and then loaded from this "uber.jar" , then we disable dd agent with ERROR log. This test is emulating this situation.

Reply:
So, any chance somebody of original authors could provide better name/docs/comments for this test?

Reply:
Sure. What you think would be the better name for this test?

Reply:
done, please review

btw, @lpriima thx for adding this guard! i had to troubleshoot a really weird error earlier this year, it did not exhibit itself as NPE like here, but ended up being same issue where customer had bundled our agent into their uber jar

Reply:
That happens several times with us as well. We got NPEs on de-referencing classloader of random application classes around user code.